src/test/metrics_test.cpp

```diff
@@ -0,0 +1,808 @@
+    EXPECT_THAT(registry.collect(), HasSubstr("name_count{other=\"data\"} 0\n"));
+}
+
+TEST(MetricsHistogram, RemoveLastMetricRemovesFamily) {
+    MetricRegistry registry;
+    auto family = registry.createFamily<MetricHistogram>("name", "desc");
+    auto metric = family->addMetric({{"label", "value"}}, {10});
+    EXPECT_THAT(registry.collect(), HasSubstr("# HELP name"));
+    EXPECT_THAT(registry.collect(), HasSubstr("# TYPE name"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_bucket{label=\"value\",le=\"10\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_bucket{label=\"value\",le=\"+Inf\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_sum{label=\"value\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_count{label=\"value\"} 0\n"));
+    family->remove(metric);
+    EXPECT_EQ(registry.collect().size(), 0);
+}
+
+TEST(MetricsHistogram, RemoveRemovedMetric) {
+    MetricRegistry registry;
+    auto family = registry.createFamily<MetricHistogram>("name", "desc");
+    auto metric = family->addMetric({{"label", "value"}});
+    family->remove(metric);
+    family->remove(metric);
+    EXPECT_EQ(registry.collect().size(), 0);
+}
+
+TEST(MetricsHistogram, RemoveEntireFamilyOfMetrics) {
+    MetricRegistry registry;
+    auto family1 = registry.createFamily<MetricHistogram>("name", "desc");
+    auto family2 = registry.createFamily<MetricHistogram>("fam", "desc");
+    auto metric1 = family1->addMetric({{"label", "value"}}, {10});
+    auto metric2 = family1->addMetric({{"other", "data"}}, {10});
+    auto metric3 = family2->addMetric({{"other", "data"}}, {10});
+    EXPECT_THAT(registry.collect(), HasSubstr("# HELP name"));
+    EXPECT_THAT(registry.collect(), HasSubstr("# TYPE name"));
+    EXPECT_THAT(registry.collect(), HasSubstr("# HELP fam"));
+    EXPECT_THAT(registry.collect(), HasSubstr("# TYPE fam"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_bucket{label=\"value\",le=\"10\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_bucket{label=\"value\",le=\"+Inf\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_sum{label=\"value\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_count{label=\"value\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_bucket{other=\"data\",le=\"10\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_bucket{other=\"data\",le=\"+Inf\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_sum{other=\"data\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("name_count{other=\"data\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("fam_bucket{other=\"data\",le=\"10\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("fam_bucket{other=\"data\",le=\"+Inf\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("fam_sum{other=\"data\"} 0\n"));
+    EXPECT_THAT(registry.collect(), HasSubstr("fam_count{other=\"data\"} 0\n"));
```

Comment:
Again - we verified that adding metrics worked in previous tests - I don;t think that we need to verify it once again

Reply:
Removed